solver/src/driver.rs

```diff
@@ -80,15 +84,29 @@ impl Driver {
             Some(settlement) => settlement,
         };
         info!("Computed {:?}", settlement);
+        let nonce = self
+            .transaction_count()
+            .await
+            .context("failed to get transaction_count")?;
         // TODO: check if we need to approve spending to uniswap
         settlement_submission::submit(
-            settlement,
             &self.settlement_contract,
+            nonce,
             self.gas_price_estimator.as_ref(),
             self.target_confirm_time,
+            GAS_PRICE_CAP,
+            settlement,
         )
         .await
         .context("failed to submit settlement")?;
         Ok(())
     }
+
+    async fn transaction_count(&self) -> Result<U256> {
```

Comment:
nit: why have this logic here and not in `submit`? Seems like it would fit more coherently over there.

Reply:
I am actually surprised how many lines are needed just to get the nonce. Also, not sure what the settlement contract has to do with an EOA's nonce.